packages/atomic/src/components/search/atomic-refine-toggle/atomic-refine-toggle.tsx

```diff
@@ -40,6 +40,10 @@ export class AtomicRefineToggle implements InitializableComponent {
   }
 
   private loadModal() {
+    if (this.modalRef) {
+      return;
```

Comment:
The smart snippet modal does the same

Reply:
I think that this logic should be owned by the creator of the modal, since from the modal's point of view, it could've been included in the HTML file (just like the modal example). Additionally, the modal isn't particularly aware of what "kind" of modal it is, it only has a reference to a container which it must hide when inactive.